src/test/java/net/snowflake/client/core/IncidentIT.java

```diff
@@ -118,7 +118,10 @@ public void simpleFlushTest() throws SQLException {
 
     Incident incident =
         new Incident(
-            connection.unwrap(SnowflakeConnectionV1.class).getSfSession(), exc, jobId, requestId);
+            (SFSession) connection.unwrap(SnowflakeConnectionV1.class).getSfSession(),
```

Comment:
why we need to cast it here?

Reply:
Can we use unwrap instead?

Reply:
I'm trying to limit classes that should only be used with the default client implementation to enforce use of the `SFSession` type. Since `connection.getSfSession()` returns a `BaseSFSession` type, I have to cast it to `SFSession`, since `Incident`'s constructor still takes an `SFSession` type.

 `unwrap` work only for classes that implement `java.sql.Wrapper`. Should we make `BaseSFSession` implement that?

Reply:
I also just had a chat with @sfc-gh-mknister. In fact, I may need to perform this cast some more in the IT tests due to the old-version testing that happens, which means it cannot use the new `BaseSFSession` type.